src/Core/Form/IdentifiableObject/DataHandler/OrderReturnFormDataHandler.php

```
@@ -0,0 +1,72 @@
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+use OrderReturn;
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\OrderReturn\Command\UpdateOrderReturnStateCommand;
+
+/**
+ * Saves or updates order return data submitted in form
+ */
+class OrderReturnFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
+    {
+        $this->commandBus = $commandBus;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($orderReturnId, array $data): void
+    {
+        $orderReturnStateId = (int) $data['order_return_state'];
+
+        $this->commandBus->handle(new UpdateOrderReturnStateCommand($orderReturnId, $orderReturnStateId));
+    }
+
+    /**
+     * Order Return doesn't have a create option
+     *
+     * @param array $data
+     */
+    public function create(array $data): void
+    {
+        @trigger_error(OrderReturn::class . ' is not created by form, this method should never be called', E_USER_WARNING);
```

throw exception here

---

Why would exception be better? Exceptions are better in general because they can be caught and this error should never be caught because create method should never be used in the first place.

---

I kind of agree with Fabien on this one, the method is never called as you said so anyway the exception should never be thrown. But if someone wrongly uses the form handler he needs to know right away that something is wrong, the `trigger_error` is a bit too silent that's why I think it's better to have an explicit exception that breaks everything so there's no ambiguity that something is not working.

We had many issues in the past because the legacy codes often fail silently, thus you don't know which step of a complex workflow has failed and need to investigate step by step to understand what went wrong. Since then I'm in favor of failing loudly and early 😅

I think we should add a `NotImplementedException extends CoreException` for those cases (there is one in the bundle namespace but we'd better use one from the core).